From 526d703a74752b66fd54ac4a9133a3de31c90549 Mon Sep 17 00:00:00 2001
From: Gleb Natapov <gleb@redhat.com>
Date: Sun, 11 Sep 2011 09:05:50 +0200
Subject: [PATCH] do not advertise S4/S3 in DSDT

RH-Author: Gleb Natapov <gleb@redhat.com>
Message-id: <20110911090550.GF21417@redhat.com>
Patchwork-id: 32631
O-Subject: [PATCHv2 RHEL6.2 SEABIOS] do not advertise S4/S3 in DSDT
Bugzilla: 736522
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>
RH-Acked-by: Marcelo Tosatti <mtosatti@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>

To compile Seabios with PM support run make "DSDT_CPP_FLAGS=-DDSDT_PM"

BZ: 736522
Upstream: local patch

Signed-off-by: Gleb Natapov <gleb@redhat.com>
--
			Gleb.

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 Makefile          |    2 +-
 src/acpi-dsdt.dsl |    2 ++
 2 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index b185ac8..f5c7104 100644
--- a/Makefile
+++ b/Makefile
@@ -186,7 +186,7 @@ $(OUT)vgabios.bin: $(OUT)vgabios.bin.raw tools/buildrom.py
 ####### dsdt build rules
 src/%.hex: src/%.dsl
 	@echo "Compiling DSDT"
-	$(Q)cpp -P $< > $(OUT)$*.dsl.i
+	$(Q)cpp $(DSDT_CPP_FLAGS) -P $< > $(OUT)$*.dsl.i
 	$(Q)iasl -tc -p $(OUT)$* $(OUT)$*.dsl.i
 	$(Q)cp $(OUT)$*.hex $@
 
diff --git a/src/acpi-dsdt.dsl b/src/acpi-dsdt.dsl
index b54c558..1db5ae9 100644
--- a/src/acpi-dsdt.dsl
+++ b/src/acpi-dsdt.dsl
@@ -722,6 +722,7 @@ DefinitionBlock (
         }
     }
 
+#ifdef DSDT_PM
     /*
      * S3 (suspend-to-ram), S4 (suspend-to-disk) and S5 (power-off) type codes:
      * must match piix4 emulation.
@@ -740,6 +741,7 @@ DefinitionBlock (
         Zero,  /* reserved */
         Zero   /* reserved */
     })
+#endif
     Name (\_S5, Package (0x04)
     {
         Zero,  /* PM1a_CNT.SLP_TYP */
-- 
1.7.4.4

